### PR TITLE
test: migrate call and callback tests

### PR DESCRIPTION
**Pull request by nechaido**
Refs: #145
Refs: #146 
Refs: #53.

### CODE REVIEW

**Comment by nechaido** [comment]
Blocked until #190.

**Comment by aqrln** [comment]
#190 landed, unblocking this.

**Comment by aqrln** [comment]
I think we should use `connection.callMethod()` in these tests, not `connection.inspect()` and remote proxies.

File: test/node/connection-call.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    );
+  }
+  );
+});
+
+test.test('must perform call that returns an error', (test) => {
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    connection.callMethod('calculator', 'divide', [10, 0], (error) => {
+      test.assert(error, 'callMethod must return an error');
+      test.equal(error.code, 1);
```
Comment (belochub) [inline test/node/connection-call.js]: You should also test the error message for equality here.

Comment (nechaido) [inline test/node/connection-call.js]: There is no need for checking error code, but I'll check error message instead

**Comment by nechaido** [comment]
@aqrln added.

File: test/fixtures/application.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const name = 'testApp';
+const expectedErrorMessage = 'Zero division';
+const interfaces = {
+  calculator: {
+    answer(connection, callback) {
+      callback(null, 42);
+    },
+    divide(connection, divided, divider, callback) {
```
Comment (lundibundi) [inline test/fixtures/application.js]: Just a nit but divided -> dividend, divider -> divisor.

File: test/node/connection-call.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+});
+
+test.test('must perform call with no arguments and no return value', (test) => {
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    connection.callMethod('calculator', 'doNothing', [], (error) => {
+      test.assertNot(error, 'callMethod must not return an error');
+      test.end();
+    });
+  }
+  );
```
Comment (lundibundi) [inline test/node/connection-call.js]: Just another nit, put them on one line? ( `});` )

File: test/node/connection-call.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+test.test('must perform call with no arguments and return value', (test) => {
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    connection.callMethod('calculator', 'answer', [], (error, result) => {
+      test.assertNot(error, 'callMethod must not return an error');
+      test.equal(result, 42);
+      test.end();
+    });
+  }
+  );
```
Comment (lundibundi) [inline test/node/connection-call.js]: ditto.

File: test/node/connection-call.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    connection.callMethod('calculator', 'divide', [20, 10],
+      (error, result) => {
+        test.assertNot(error, 'callMethod must not return an error');
+        test.equal(result, 2);
+        test.end();
+      }
+    );
+  }
+  );
```
Comment (lundibundi) [inline test/node/connection-call.js]: ditto.

File: test/node/connection-call.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+test.test('must perform call that returns an error', (test) => {
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    connection.callMethod('calculator', 'divide', [10, 0], (error) => {
+      test.assert(error, 'callMethod must return an error');
+      test.equal(error.message, `Error: ${app.expectedErrorMessage}`);
+      test.end();
+    });
+  }
+  );
```
Comment (lundibundi) [inline test/node/connection-call.js]: ditto.

File: test/node/connection-call.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    );
+  }
+  );
+});
+
+test.test('must perform call that returns an error', (test) => {
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    connection.callMethod('calculator', 'divide', [10, 0], (error) => {
+      test.assert(error, 'callMethod must return an error');
+      test.equal(error.message, `Error: ${app.expectedErrorMessage}`);
```
Comment (lundibundi) [inline test/node/connection-call.js]: It's better to compare it as new Error(...).message.

**Comment by nechaido** [comment]
@lundibundi fixed.

**Comment by nechaido** [comment]
@aqrln `test.end()` will ensure it, because it will be called only in the callback and the test will not end until meeting quota, which is not set, or calling `test.end()`.

**Comment by nechaido** [comment]
Landed in 8bbb4792cba5caf63a6e3e26e7049a26a22dd082.

**Pull request closed without merging**